### DISCUSSION TITLE
Using HTTP connection pool for HTTP/S3

### DISCUSSION THREAD

**Discussion by kesavkolla** [duckdb/duckdb#4821] [Ideas]
It will be good idea to keep a connection pool for HTTP so that standard TLS handshakes and DNS requests are not repeated per every request. Usually people keep parquet files in S3 and would want to use duckdb for querying. It would be great performance benefit if we can re-use the HTTP connections.

**Comment by hannes** [comment]
This is indeed also something we may consider at some point

**Comment by naah69** [comment]
https://github.com/duckdb/duckdb-httpfs/issues/82

@hannes 
We had the same need!
There are too many tcp connection with `TIME_CLOSE`.

```bash
root@label-8cfccb645-268xb:/app# netstat -apto | grep 100.125.160.3 | grep WAIT | wc -l
5331
```
